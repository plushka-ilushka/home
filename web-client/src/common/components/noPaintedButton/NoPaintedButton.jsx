import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './noPaintedButton.scss';

export default function NoPaintedButton(props) {
  const {
    className,
    onClick,
    caption,
    ...setProps
  } = props;

  return (
    <button
      className={classNames('no-painted-button', className)}
      type="button"
      onClick={onClick}
      {...setProps}
    >
      {caption}
    </button>
  );
}

NoPaintedButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.bool,
  caption: PropTypes.string,
};

NoPaintedButton.defaultProps = {
  className: undefined,
  caption: undefined,
  onClick: undefined,
};
