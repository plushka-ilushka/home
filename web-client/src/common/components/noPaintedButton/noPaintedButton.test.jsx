import React from 'react';
import { mount } from 'enzyme';
import NoPaintedButton from './NoPaintedButton';

describe('NoPaintedButton Component', () => {
  it('NoPaintedButton Snapshot', () => {
    expect(mount(<NoPaintedButton />)).toMatchSnapshot();
  });
});
