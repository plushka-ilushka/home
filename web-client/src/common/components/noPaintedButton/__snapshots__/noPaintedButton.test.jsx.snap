// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPaintedButton Component NoPaintedButton Snapshot 1`] = `
<NoPaintedButton>
  <button
    className="no-painted-button"
    type="button"
  />
</NoPaintedButton>
`;
